junifer/markers/crossatlas_functional_connectivity.py

```diff
@@ -0,0 +1,164 @@
+"""Provide marker class to calculate cross-atlas FC."""
+
+# Authors: Leonard Sasse <l.sasse@fz-juelich.de>
+#          Kaustubh R. Patil <k.patil@fz-juelich.de>
+# License: AGPL
+
+from typing import Any, Dict, List, Optional
+
+import pandas as pd
+
+from ..api.decorators import register_marker
+from ..storage import BaseFeatureStorage
+from ..utils import logger
+from .base import BaseMarker
+from .parcel_aggregation import ParcelAggregation
+from .utils import _correlate_dataframes
+
+
+@register_marker
+class CrossAtlasFC(BaseMarker):
+    """Class for calculating parcel-wise correlations between two atlases.
+
+    Parameters
+    ----------
+    atlas_one : str
+        The name of the first atlas.
+    atlas_two : str
+        The name of the second atlas.
+    aggregation_method : str, optional
+        The aggregation method (default "mean").
+    correlation_method : str, optional
+        Any method that can be passed to:
+        https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.corr.html
+        (default "pearson").
+    name : str, optional
+        The name of the marker. If None, will use the class name
+        (default None).
+    """
+
+    def __init__(
+        self,
+        atlas_one: str,
+        atlas_two: str,
+        aggregation_method: str = "mean",
+        correlation_method: str = "pearson",
+        name: str = None,
+    ) -> None:
+        """Initialize the class."""
+        self.atlas_one = atlas_one
+        self.atlas_two = atlas_two
+        self.aggregation_method = aggregation_method
+        self.correlation_method = correlation_method
+        super().__init__(on=["BOLD"], name=name)
+
+    def get_valid_inputs(self) -> List[str]:
+        """Get valid data types for input.
+
+        Returns
+        -------
+        list of str
+            The list of data types that can be used as input for this marker
+
+        """
+        return ["BOLD"]
+
+    def get_output_kind(self, input: List[str]) -> List[str]:
+        """Get output kind.
+
+        Parameters
+        ----------
+        input : list of str
+            The input to the marker. The list must contain the
+            available Junifer Data dictionary keys.
+
+        Returns
+        -------
+        list of str
+            The updated list of output kinds, as storage possibilities.
+
+        """
+        return ["matrix"]
+
+    def store(
+        self,
+        kind,
+        out: Dict[str, Any],
+        storage: "BaseFeatureStorage",
+    ) -> None:
+        """Store.
+
+        Parameters
+        ----------
+        kind : {"BOLD"}
+            The data kind to store.
+        out : dict
+            The computed result as a dictionary to store.
+        storage : storage-like
+            The storage class, for example, SQLiteFeatureStorage.
+
+        """
+        logger.debug(f"Storing BOLD-based marker in {storage}")
```

Comment:
For consistency, shall we keep `... BOLD in ...`?

Reply:
I am a bit uncomfortable with that as this will suggest to  a user that a BOLD signal is being saved, which is always a time series.

Reply:
what say you @fraimondo